### PR TITLE
LinkedList palindrome using stack and reverse linked list

### PR DESCRIPTION
**Pull request by vidhyaa12**
@jaisong87

### CODE REVIEW

**Comment by vidhyaa12** [comment]
Linked list palindrome without using stack (through linked list reverse strategy coming soon)

File: answers/src/main/java/com/coding/solutions/basic/LinkedListPalindrome.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            return true;
+        }
+
+        int left = 0;
+        int right = inputList.size() / 2;
+        if (inputList.size() % 2 != 0) {
+            right = (inputList.size() / 2) + 1;
+        }
+
+        Stack<Character> stack = new Stack<Character>();
```
Comment (jaisong87) [inline answers/src/main/java/com/coding/solutions/basic/LinkedListPalindrome.java:21]: Do this in place. Don't use stack. Also don't use Character. Make it templatized.

File: answers/src/main/java/com/coding/solutions/basic/SinglyLinkedList.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        while(true){
+            if(tmp == null){
+                break;
+            }
+            System.out.println(tmp.getValue());
+            tmp = tmp.getNext();
+        }
+    }
+
+    public void reverse() {
```
Comment (jaisong87) [inline answers/src/main/java/com/coding/solutions/basic/SinglyLinkedList.java:119]: Write a reverse function that can reverse subList from some node position N, and use it in your palindrome problem.

Comment (vidhyaa12) [inline answers/src/main/java/com/coding/solutions/basic/SinglyLinkedList.java:119]: This okay ?

``` java
 public SinglyLinkedList reverseSubListFromStart(int start) {
        if (this == null || this.head == null) {
            return null;
        }

        SinglyLinkedListNode prev = null;
        SinglyLinkedListNode current = head;

        int index = 0;
        while (index < start) {
            current = current.getNext();
            index++;
        }

        SinglyLinkedListNode next = ( current != null) ? current.getNext() : null;
        SinglyLinkedListNode nextToNext = (next != null) ? next.getNext() : null;

        while ( current != null ) {
//            System.out.println(print(prev) + " -> " + print(current) + " -> " + print(next) + " -> " + print(nextToNext));
            current.setNext(prev);
            prev = current;
            current = next;
            next = nextToNext;
            if (nextToNext != null) nextToNext = nextToNext.getNext();
        }
        head = prev;

        current = head;
        SinglyLinkedList result = new SinglyLinkedList();
        while (current != null) {
            result.add(current.getValue());
            current = current.getNext();
        }
        return result;
    }
```

Comment (vidhyaa12) [inline answers/src/main/java/com/coding/solutions/basic/SinglyLinkedList.java:119]: And to check if a LinkedList is a palindrome I can use the equals method to check for equality of the 2 sublists (reversed and remaining part)

**Pull request merged**